File: src/main/java/fi/finlit/edith/domain/LinkElement.java
/**
 *
 */
package fi.finlit.edith.domain;


public class LinkElement extends Identifiable implements ParagraphElement {

    private String string;

    private String reference;

    public LinkElement() {
    }

    public LinkElement(String string) {
        this.string = string;
    }

    @Override
    public ParagraphElement copy() {
        LinkElement element = new LinkElement(string);
        element.setReference(reference);
        return element;
    }

    @Override
    public String toString() {
        return "<bibliograph" + (reference == null ? "" : " ref=\"" + reference + "\"") + ">"
                + string + "</bibliograph>";
    }

    public void setReference(String reference) {
        this.reference = reference;
    }

    public String getReference() {
        return reference;
    }

    public String getString() {
        return string;
    }
}

File: src/main/scala/fi/finlit/edith/sql/domain/LinkElement.scala
package fi.finlit.edith.sql.domain

import scala.reflect.BeanProperty
//remove if not needed
import scala.collection.JavaConversions._

class LinkElement(@BeanProperty val string: String) extends ParagraphElement {

  @BeanProperty
  var reference: String = _

  override def copy(): ParagraphElement = {
    var element = new LinkElement(string)
    element.setReference(reference)
    element
  }

  override def toString(): String = {
    "<bibliograph" + (if (reference == null) "" else " ref=\"" + reference + "\"") + ">" + string + "</bibliograph>"
  }
}
